lib/templates/nuxt-content.js

```diff
@@ -37,7 +67,42 @@ function processNode (node, h, doc) {
     return node.value
   }
 
-  return h(node.tag, propsToData(node.props, doc), node.children.map((child) => processNode(child, h, doc)))
+  const slotData = slotsToData(node || {}, h, doc)
+  const propData = propsToData(node.props, doc)
+  const data = Object.assign({}, slotData, propData)
+
+  /**
+   * Process child nodes, flat-mapping templates pointing to default slots.
+   */
+  const children = []
+  for (const child of node.children) {
+    if (isTemplate(child) && !isDefaultTemplate(child)) { continue }
+
+    const processQueue = isDefaultTemplate(child) ? child.content: [child]
+    children.push(...processQueue.map((node) => processNode(node, h, doc)))
+  }
+
+  return h(node.tag, data, children)
+}
+
+const DEFAULT_SLOT = 'default'
+
+function isDefaultTemplate(node) {
+  return isTemplate(node) && getSlotName(node) === DEFAULT_SLOT
+}
+
+function isTemplate(node) {
+  return node.tag === 'template'
+}
+
+function getSlotName (node) {
+  let name = ''
+  for (const propName of Object.keys(node.props)) {
+    if (!propName.startsWith('v-slot:')) { return }
```

Comment:
What about checking if it starts with ‘#’ as well to handle the shortcut?

Reply:
For some reason, I thought `rehype` did not parse it. It does. I'll change. Thanks for the heads up.